src/main/java/model/path/RandomPathGenerator.java

```diff
@@ -0,0 +1,34 @@
+package model.path;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class RandomPathGenerator implements PathGenerator {
+    private final Random random = new Random();
+
+    @Override
+    public List<Path> generate(final int count) {
+        final List<Path> paths = new ArrayList<>();
+        while (paths.size() < count) {
+            Path randomPath = getNextPath(paths);
+            paths.add(randomPath);
+        }
+        return paths;
+    }
+
+    private Path getNextPath(final List<Path> paths) {
+        if (!paths.isEmpty() && getLastPath(paths) == Path.EXIST) {
```

Comment:
아래와 같이 메서드를 추출하면 한눈에 알 수 있을것같아요
물론 지금도 읽기에 무리는 없어보이네요 ㅎㅎ.... 넘어가셔도됩니다

```suggestion
if (!paths.isEmpty() && this.existLastPath(paths)) {
```